CMakeLists.txt

```diff
@@ -0,0 +1,40 @@
+cmake_minimum_required(VERSION 3.5)
+project(nlohmann_json_schema_validator_vendor)
+
+find_package(ament_cmake REQUIRED)
+
+option(FORCE_BUILD_VENDOR_PKG
+  "Build nlohmann_json_schema_validator from source, even if system-install package is available"
+  OFF
+)
+
+if(NOT FORCE_BUILD_VENDOR_PKG)
+  find_package(nlohmann_json_schema_validator QUIET)
+endif()
+
+
+if(NOT nlohmann_json_schema_validator_FOUND)
+
+  set(setup_install_dir "${CMAKE_CURRENT_BINARY_DIR}/nlohmann_json_schema_validator")
+
+  include(ExternalProject)
+  ExternalProject_Add(nlohmann_json_schema_validator
```

Comment:
I remember @marcoag mentioning something about having issues with packaging libraries that need to be fetched from the internet using `FetchContent` and cousins. I might be completely wrong about this but I'll just tag him on this PR as this to be doubly sure as this is not my area of expertise.